typing/jkind_types.ml

```diff
@@ -732,37 +717,34 @@ module Layout_and_axes = struct
     match f t.layout with None -> None | Some layout -> Some { t with layout }
 
   let map_type_expr f t =
-    { t with upper_bounds = Bounds.map_type_expr f t.upper_bounds }
+    { t with with_bounds = With_bounds.map_type_expr f t.with_bounds }
 
   let equal eq_layout { layout = lay1; upper_bounds = bounds1 }
       { layout = lay2; upper_bounds = bounds2 } =
     eq_layout lay1 lay2 && Bounds.equal bounds1 bounds2
 
-  let try_allow_l { layout; upper_bounds } =
-    let module Bounds_map_option = Bounds.Map.Monadic (Misc.Stdlib.Monad.Option) in
-    match
-      Bounds_map_option.f
-        { f = (fun ~axis:_ bound -> Bound.try_allow_l bound) }
-        upper_bounds
-    with
-    | Some upper_bounds -> Some { layout; upper_bounds }
+  let try_allow_l :
+      type l r.
+      ('type_expr, 'layout, l * r) t ->
+      ('type_expr, 'layout, Allowance.allowed * r) t option =
+   fun { layout; upper_bounds; with_bounds } ->
+    match With_bounds.try_allow_l with_bounds with
     | None -> None
+    | Some with_bounds ->
+      Some { layout; upper_bounds = Obj.magic upper_bounds; with_bounds }
```

Comment:
Yikes! Why the `Obj.magic`? At the least, this would need a loud comment. But I think it's not necessary?

Reply:
oh, yikes indeed. I had intended to remove this entirely but missed it when reviewing.